### ISSUE TITLE
Function composition specialised to a particular type doesn't work properly

### ISSUE THREAD

**Issue by nick8325** [nick8325/quickspec#35]
With the following example we don't discover `(f . g) x = f (g x)`.

```
{-# LANGUAGE TypeApplications #-}
import QuickSpec

main = quickSpec [
  con "." ((.) @Char @Char @Char) ]
```

Problem is that the most general type of `g x` is `a -> a`, which  is not in the type universe.

One solution is to close the type universe under generalisation (replacing types with a type variable). However, we do not want e.g. `Char -> a` to be in the type universe as it is not useful for testing the signature.

Instead, we maybe want to make `inUniverse` take a polymorphic type and check if it can be instantiated to something in the universe.

Wait until we have a test suite to fix this.

**Comment by nick8325** [comment]
Actually, this seems to be about type variable that occur in the term but not in its type. In this case, the type of `g x` is `b` where `g :: a -> b`, and `Polymorphic.hs` never tries instantiating `a`.

**Comment by nick8325** [comment]
Fixed by b0aa67c.
